packages/@ember/-internals/routing/lib/system/route.ts

```diff
@@ -278,8 +278,8 @@ class Route<T = unknown>
     super(owner);
 
     if (owner) {
-      let router = owner.lookup<EmberRouter>('router:main');
-      let bucketCache = owner.lookup<BucketCache>(P`-bucket-cache:main`);
+      let router = owner.lookup('router:main') as EmberRouter;
```

Comment:
**~:** Same comment throughout this file of course. It's an improvement regardless.

All of these plus a conversation with @dfreeman earlier have me thinking: we *should* be able to provide an internal registry extension *that isn't published* which makes these all work. That's good follow-on which I'm happy to advise about. (We can make that work with the microsyntax using template literal types.)